AutoCollection/HttpDependencies.ts

```diff
@@ -122,31 +122,33 @@ class AutoCollectHttpDependencies {
         // methods exist before invoking them.
         if (Util.canIncludeCorrelationHeader(client, requestParser.getUrl()) && telemetry.request.getHeader && telemetry.request.setHeader) {
             if (client.config && client.config.correlationId) {
-                const correlationHeader = telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader);
-                if (correlationHeader) {
-                    const components = correlationHeader.split(",");
-                    const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
-                    if (!components.some((value) => value.substring(0,key.length) === key)) {
-                        telemetry.request.setHeader(
-                            RequestResponseHeaders.requestContextHeader,
-                            `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
-                    }
+                // getHeader returns "any" type in newer versions of node. In basic scenarios, this will be <string | string[] | number>, but could be modified to anything else via middleware
+                const correlationHeader = <any>telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader)
+                if (typeof correlationHeader === "string") {
+                    AutoCollectHttpDependencies.addRequestCorrelationIdHeaderFromString(client, telemetry, correlationHeader)
+                } else if (correlationHeader instanceof Array) { // string[]
+                    const headers = correlationHeader.join(",");
+                    AutoCollectHttpDependencies.addRequestCorrelationIdHeaderFromString(client, telemetry, headers)
+                } else if (correlationHeader && typeof (correlationHeader as any).toString === "function") {
+                    // best effort attempt: requires well-defined toString
```

Comment:
AutoCollectHttpDependencies.addRequestCorrelationIdHeaderFromString(..) is being called multiple times in this block.  Try to set a single header string for various cases and just call AutoCollectHttpDependencies.addRequestCorrelationIdHeaderFromString(..) once with the header value.